xarray/backends/scipy_.py

```diff
@@ -55,6 +55,18 @@ def __getitem__(self, key):
             copy = self.datastore.ds.use_mmap
             return np.array(data, dtype=self.dtype, copy=copy)
 
+    def __setitem__(self, key, value):
+        with self.datastore.ensure_open(autoclose=True):
+            data = self.get_array()
```

Comment:
This needs to be `self.datastore.ds.variables[self.variable_name]` (a `netcdf_variable` object), not `self.get_array()`, which returns the `.data` (a numpy array). You can't enlarge a numpy array, but *can* enlarge a scipy netcdf variable.

(This manifests itself in writing a netcdf file with a time dimension of length 0. Xarray then crashes when attempting to a decode a length 0 time variable, which is an unrelated bug.)

Reply:
Thanks @shoyer - this appears to fix the scipy issue.